core/src/test/java/org/apache/calcite/plan/RelWriterTest.java

```diff
@@ -706,6 +717,51 @@
             + "    LogicalTableScan(table=[[hr, emps]])\n"));
   }
 
+  @Test void testJsonToRex() throws JsonProcessingException {
+    // Test simple literal without inputs
+    final String jsonString1 = "{\n"
+        + "            \"literal\": 10,\n"
+        + "            \"type\": {\n"
+        + "              \"type\": \"INTEGER\",\n"
+        + "              \"nullable\": false\n"
+        + "            }\n"
+        + "          }\n";
+
+    assertReadRex(jsonString1, "10");
+
+    // Test Binary with an input
+    final String jsonString2 = "{ \"op\": \n"
+        + "      { \"name\": \"+\",\n"
+        + "        \"kind\": \"PLUS\",\n"
+        + "        \"syntax\": \"BINARY\"\n"
+        + "      },\n"
+        + "      \"operands\": [\n"
+        + "        {\n"
+        + "          \"input\": 1,\n"
+        + "          \"sql\": \"column + 1\"\n"
+        + "        },\n"
+        + "        {\n"
+        + "          \"literal\": 1,\n"
+        + "          \"type\": { \"type\": \"INTEGER\", \"nullable\": false }\n"
+        + "        }\n"
+        + "      ]\n"
+        + "    }";
+    assertReadRex(jsonString2, "+(1, 1)");
+  }
+
+  private void assertReadRex(String jsonString1, String expected)
```

Comment:
better name would be `assertThatReadExpressionResult`

change parameter to from `String` to `Matcher<String>`. then there will be an `is` at the call site, and the code will read somewhat like English (per junit best practices, e.g. `assertThat(1 + 2, is(3));`